### ISSUE TITLE
xf86-intel-video tearing with compton

### ISSUE THREAD

**Issue by edeustua** [chjj/compton#376]
Hi!

I've recently started to have problems with compton and the xf86-intel-video driver.

I tracked the issue down to this commit in the xf86-intel-video project.

```
commit 1f6dfc9df6780bd3768bf065156b6e8dae05b80c
Author: Chris Wilson <chris@chris-wilson.co.uk>
Date:   Tue Aug 2 09:06:50 2016 +0100

    sna: Only flush GPU bo for a damage event

    Based on xf86-video-ati

    commit 9a1afbf61fbb2827c86bd86d295fa0848980d60b
    Author: Michel Dänzer <michel.daenzer@amd.com>
    Date:   Mon Jul 11 12:22:09 2016 +0900

        Use EventCallback to avoid flushing every time in the FlushCallback

    reports seeing an improvement in reducing flushes at the expense of
    checking every event for a DamageNotifyEvent. Since we also mix
    rendering with SHM buffers, we have a more diverse set of conditions
    under which to flush - but maybe we will see enough of a win for DRI to
    merit. So far seeing improvement of ~20% for series of small operations
    under the compositor without seeing any regressions, should benefit
    composited desktop users. The biggest danger here is missed flushes.

    Signed-off-by: Chris Wilson <chris@chris-wilson.co.uk>
```

The problem appears to be, mostly, in the rendering of urxvt and xterm. I get a lot of tearing, and also, the cursor disappears erratically.

I'm guessing that compton is not able to use the `EventCallBack` function for rendering certain windows, or something similar, and that is why I'm getting this issue. Is this true? And if not is there maybe another way of fixing this?

Thank you for your time,

Emiliano

**Comment by edeustua** [comment]
Apparently `--xrender-sync-fenc` or `--xrender-sync` fixes the problem.

**Comment by shioyama** [comment]
Noticed this problem as well with urxvt, and can confirm that `--xrender-sync-fence` fixes the issue for me.

**Comment by shioyama** [comment]
Added note: `--xrender-sync` alone doesn't seem to do it.

**Comment by iammer** [comment]
I am seeing this as well.  But, neither `--xrender-sync-fence` or `--xrender-sync` together or separately are helping.  I currently am only using compton to stop tearing for videos.  So, I have just been killing it when using the terminal.

**Comment by edeustua** [comment]
@iammer Can you share your `.compton.conf` file?

**Comment by iammer** [comment]
After some research, I think I figured it out.  I am not using a .compton.conf file.  I just had i3 run `/usr/bin/compton --xrender-sync-fence --xrender-sync --backend glx --vsync opengl-swc -b` at startup.  I realized the xrender options probably do nothing for the glx backend.  Switching to the xrender backend did fix the urxvt tearing but didn't fix my screen tearing for video.  Fortunately, xr_glx_hybrid seems to fix both.

**Comment by ngleb** [comment]
The issues with xterm seems to be fixed with the proposed options, but I have tearing in Firefox in fullscreen HTML5. It doesn't happen on latest beta version of compton which is stable in Gentoo. Is anyone also have similar problem?

**Comment by ff2000** [comment]
I also had severe issues with the intel driver. I didn't manage to workaround this with all the proposed config-tweaks. The only solution was using glamor, which unfortunately had a big impact on performance, especially in firefox. Yesterday I installed the most recent intel-driver snapshot offered by Gentoo (2.99.917_p20170216) and until now I did not see any issue with tearing, and performance is really nice. So probably they reverted the change that caused the issues with compton or they found a way to prevent it.
Can anybody confirm my observations?
